File: SPPIDDataLayer/SPPIDDataLayer/Configuration.vb
Imports System.Collections.Generic
Imports System.Xml.Serialization
Imports System.Runtime.Serialization
Imports System.Linq
Imports System.Text

Imports org.iringtools.library


<DataContract(Name:="document")> _
Public Class SPPIDConfiguration
    <DataMember(Name:="siteConnectionString", Order:=0)> _
    Public Property SiteConnectionString() As String
        Get
            Return m_SiteConnectionString
        End Get
        Set(value As String)
            m_SiteConnectionString = value
        End Set
    End Property
    Private m_SiteConnectionString As String

    <DataMember(Name:="plantConnectionString", Order:=1)> _
    Public Property PlantConnectionString() As String
        Get
            Return m_PlantConnectionString
        End Get
        Set(value As String)
            m_PlantConnectionString = value
        End Set
    End Property
    Private m_PlantConnectionString As String

    <DataMember(Name:="stagingConnectionString", Order:=2)> _
    Public Property StagingConnectionString() As String
        Get
            Return m_StagingConnectionString
        End Get
        Set(value As String)
            m_StagingConnectionString = value
        End Set
    End Property
    Private m_StagingConnectionString As String





End Class

File: Libraries/iRINGLibrary/Configuration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;
using System.Runtime.Serialization;

namespace org.iringtools.library
{
    [XmlRoot(ElementName = "configuration")]
    [DataContract(Name = "configuration", Namespace = "http://www.iringtools.org/library")]
    public class Configuration
    {
        [XmlElement(ElementName = "appSettings")]
        [DataMember(Name = "appSettings", Order = 0, EmitDefaultValue = false)]
        public AppSettings AppSettings { get; set; }
    }

    [DataContract(Name = "appSettings", Namespace = "http://www.iringtools.org/library")]
    public class AppSettings
    {
        [XmlElement(ElementName = "add")]
        [DataMember(Name = "add", Order = 0, EmitDefaultValue = false)]
        public List<Setting> Settings { get; set; }
    }

    [DataContract(Name = "setting", Namespace = "http://www.iringtools.org/library")]
    public class Setting
    {
        [XmlAttribute(AttributeName = "key")]
        [DataMember(Name = "key", Order = 0, EmitDefaultValue = false)]
        public String Key { get; set; }

        [XmlAttribute(AttributeName = "value")]
        [DataMember(Name = "value", Order = 1, EmitDefaultValue = false)]
        public String Value { get; set; }
    }
}
